lib/bamboo/adapters/send_grid_helper.ex

```diff
@@ -161,6 +164,32 @@ defmodule Bamboo.SendGridHelper do
     raise "expected bypass_list_management parameter to be a boolean, got #{enabled}"
   end
 
+  @doc """
+  Instruct SendGrid to enable or disable Google Analytics tracking, and
+  optionally set the UTM parameters for it. This is useful if you need to
+  control UTM tracking parameters on an individual email basis.
+
+  ## Example
+
+      email
+      |> with_google_analytics(true, %{utm_source: "email", utm_campaign: "campaign"})
+
+      email
+      |> with_google_analytics(false)
+  """
+  def with_google_analytics(email, enabled, utm_params \\ %{})
+  def with_google_analytics(email, enabled, utm_params) when is_boolean(enabled) and is_map(utm_params) do
```

Comment:
What would you think about adding a couple of unit tests for this function that verify for example the right types being passed in and that UTM params not in white list would get filtered out?